### PR TITLE
PLAT-3775 - Added support to friendly names and mailto links

### PR DESCRIPTION
**Pull request by luanapp**

### CODE REVIEW

File: src/main/java/org/symphonyoss/symphony/messageml/markdown/MarkdownParser.java [hunk 1]
```diff
@@ -244,7 +244,13 @@ private void visit(KeywordNode keyword) {
 
   private void visit(MentionNode mention) {
     try {
-      Mention node = new Mention(parent, ++index, mention.getUid(), dataProvider);
+      Mention node;
+      if (StringUtils.isEmpty(mention.getPrettyName())) {
```
Comment (ldrozdz) [inline src/main/java/org/symphonyoss/symphony/messageml/markdown/MarkdownParser.java]: This code will never be hit. Here's how Markdown parsing works:

- first out MD parser looks for the JSON node _entities_ (_MarkdownParser.enrichMarkdown()_)
- if found, it injects special sections with metadata from the entity into the original message. In the case of mentions, an "enriched" message will look like this: "Hello USER_FOLLOW1!". This is very hacky, but necessary.
- then, the resulting "enriched" message is passed onto the CommonMark parser, which calls our "visit" method to construct an MD tree

In the case of mentions, the "USER_FOLLOW" special section will be recognised as a "MentionNode" and turned into a MML "Mention", which will try to resolve the user based on their userId.

In short, either modify the "enrichMarkdown" methods (danger! beware of dragons!) or leave the MD parser as is.

File: src/test/java/org/symphonyoss/symphony/messageml/elements/MentionTest.java [hunk 2]
```diff
@@ -177,12 +177,16 @@ public void testSoftMentionInvalidEmail() throws Exception {
     String invalidAttr = "<messageML>Hello <mention email=\"invalid@email.com\" strict=\"false\" />!</messageML>";
     context.parseMessageML(invalidAttr, null, MessageML.MESSAGEML_VERSION);
 
+    String expectedJson = "{\"urls\":[{\"text\":\"mailto:invalid@email.com\","
```
Comment (ldrozdz) [inline src/test/java/org/symphonyoss/symphony/messageml/elements/MentionTest.java]: Please refer to https://github.com/symphonyoss/messageml-utils/pull/66/files for additional unit tests.
